tests/test_initrd.py

```diff
@@ -124,6 +124,47 @@ def test_extract_modules_from_deb(self):
         assert os.path.isfile(os.path.join(
             self.generator.config.target_dir, 'lib', 'modules', kversion, module))
 
+    @pytest.mark.requires_download
+    def test_extract_compressed_modules_from_deb(self, tmp_path: Path):
+        """ Test compressed modules. """
+        test_dir = os.path.dirname(os.path.abspath(__file__))
+        yaml = os.path.join(test_dir, 'data', 'initrd_noble.yaml')
+        # Prepare generator
+        generator = InitrdGenerator(yaml, str(tmp_path))
+
+        vd = VersionDepends(
+            name='linux-modules-6.8.0-31-generic',
+            package_relation=None,
+            version_relation=None,
+            version=None,
+            arch=generator.config.arch
+        )
+        package = generator.config.proxy.find_package(vd)
+        assert package
+
+        pkg = generator.config.proxy.download_package(
+            generator.config.arch, package)
+        assert pkg
+        assert pkg.local_file
+        assert os.path.isfile(pkg.local_file)
+
+        mods_temp = tempfile.mkdtemp()
+
+        pkg.extract(mods_temp)
+
+        module = 'veth'
+        generator.config.modules = [module]
+
+        kversion = generator.find_kernel_version(mods_temp)
+        assert kversion
```

Comment:
This also shouldn't use mkdtemp....

Something like that should work instead -> i.e. put it in a directory under tmp_path
```suggestion
        mods_temp_path  = tmp_path / "mods_extract"
        mods_temp_path .mkdir()
        mods_temp = str(mods_temp_path)
```

Reply:
The test works and I have other stuff to do than polishing tests.

Reply:
Alright, I'll do a cleanup round later, to get rid of all mkdtemp-usages